test/testdata/validator_with_custom_function/template.invalid_function.compile.err

```diff
@@ -0,0 +1,6 @@
+ERROR: ../../test/testdata/validator_with_custom_function/template.invalid_function.yaml:30:23: undeclared reference to 'invalidFunc' (in container '')
+ |   - match: invalidFunc(rule.port) < 0
+ | ......................^
+ERROR: ../../test/testdata/validator_with_custom_function/template.invalid_function.yaml:31:15: undeclared reference to 'invalidFunc' (in container '')
+ |     message: "invalidFunc"
+ | ..............^
```

Comment:
Ideally, the `invalidFunc` should not be tripping the validation failure here since message is expected to be a string. Perhaps, you could create a separate issue to investigate for why this occurs? In general, the check for an `expr` or `string` here is supposed to always fall back to treating the value as a string.

Reply:
Looks like the `invalidFunc` is taken bare instead of being quoted: 
https://github.com/google/cel-policy-templates-go/blob/844c9d8cff23bf7ecc706314ed255f618c99b12e/policy/compiler/compiler.go#L541

You could try compiling the `message` value and if the compilation fails then using a `%q` instead of a `%s` as this will ensure the string is properly quoted. Like I said, it's a different bug for sure, but one worth looking into.

Reply:
Ack, we will fix this in a different PR